### PR TITLE
Refactor ReactFragment testing to only allow ReactNode children

### PR DESCRIPTION
**Pull request by marisbest2**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by ericanderson** [comment]
LGTM (assuming all the tests pass), @johnnyreilly was there any reason we had the `{}` in there from 3 years ago? I can't tell why it was added nor can I think of a reason why it should be there.

**Review by ericanderson — APPROVED** [review]

**Comment by marisbest2** [comment]
@ericanderson looks like this doesn't actually work. Holding off for now, until we find a better way around it. Looks like allowing arbitrary `children` is complicated.

**Pull request closed without merging**
